File: EQueue/Rey.EQueue.Core/Entities/Teacher.cs
using Rey.EQueue.Shared;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rey.EQueue.Core.Entities
{
    public class Teacher : Entity
    {
        public Teacher()
        {
        }

        public Teacher(
            string firstName,
            string lastName,
            string? middleName = null,
            string? description = null,
            string? note = null)
        {
            FirstName = firstName;
            MiddleName = middleName;
            LastName = lastName;
            Description = description;
            Note = note;
        }

        public string FirstName { get; set; } = null!;

        public string? MiddleName { get; set; }

        public string LastName { get; set; } = null!;

        public string? Description { get; set; }

        public string? Note { get; set; }

        public ICollection<SubjectInstanceTeacher> SubjectInstanceTeachers { get; set; } = null!;
    }
}

File: EQueue/Rey.EQueue.Web/ClientApp/src/app/models/teacher.ts
export interface Teacher
{
    id: number;
    firstName: String;
    middleName: String;
    lastName: String;
    description: String;
    note: String;
}

export interface TeacherList
{
    teachers: Teacher[];
}

export interface AddTeacherModel
{
    firstName: string;
    lastName: string;
    middleName: string | null;
    description: string | null;
    note: string | null;
}